sentence_transformers/cross_encoder/losses/ListMLELoss.py

```diff
@@ -0,0 +1,313 @@
+            else:
+                # Use the original input order, assuming it's already ordered by relevance
+                sorted_logits = list_logits
+            
+            # Compute log-likelihood using Plackett-Luce model
+            scores = sorted_logits.exp()
+            cumsum_scores = torch.flip(torch.cumsum(torch.flip(scores, [0]), 0), [0])
+            log_probs = sorted_logits - torch.log(cumsum_scores + self.eps)
+            
+            # Apply position-aware lambda weights if specified
+            if self.lambda_weight is not None:
+                ranks = torch.arange(1, curr_list_size + 1, device=self.model.device)
+                position_weights = self.lambda_weight(ranks, curr_list_size)
+                log_probs = log_probs * position_weights
+            
+            # Sum the log probabilities
+            per_list_losses.append(torch.sum(-log_probs))
+        
+        if not per_list_losses:
+            return torch.tensor(0.0, device=self.model.device, requires_grad=True)
+            
+        # Average loss over all lists
+        return torch.mean(torch.stack(per_list_losses))
+
+    def get_config_dict(self) -> dict[str, float | int | str | None]:
+        """
+        Get configuration parameters for this loss function.
+
+        Returns:
+            Dictionary containing the configuration parameters
+        """
+        return {
+            "lambda_weight": None if self.lambda_weight is None else "ListMLELambdaWeight",
+            "activation_fct": fullname(self.activation_fct),
+            "mini_batch_size": self.mini_batch_size,
+            "respect_input_order": self.respect_input_order,
+        }
+
+    @property
+    def citation(self) -> str:
+        return """
+@inproceedings{lan2013position,
+    title={Position-aware ListMLE: a sequential learning process for ranking},
+    author={Lan, Yanyan and Guo, Jiafeng and Cheng, Xueqi and Liu, Tie-Yan},
+    booktitle={Proceedings of the Twenty-Ninth Conference on Uncertainty in Artificial Intelligence},
+    pages={333--342},
+    year={2013}
+}
+"""
+
+
+def create_p_list_mle_lambda_weight(list_size=None):
+    """
+    Creates a ListMLELambdaWeight with the formulation from Position-Aware ListMLE paper.
+    
+    This implements the weighting scheme from the "Position-Aware ListMLE" paper (Lan et al.).
+    The weight for each position is 2^(list_size - rank) - 1, giving higher weight to higher positions.
+    
+    Args:
+        list_size (int, optional): Size of the input list. If None, dynamically determined at runtime.
+        
+    Returns:
+        A ListMLELambdaWeight for Position-Aware ListMLE.
+    """
+    def rank_discount_fn(ranks):
+        if list_size is None:
+            curr_list_size = len(ranks)
+        else:
+            curr_list_size = list_size
+        return torch.pow(2., curr_list_size - ranks) - 1.
+        
+    return ListMLELambdaWeight(rank_discount_fn=rank_discount_fn)
```

Comment:
This is not used anywhere currently right? Am I correct in understanding that running `create_p_list_mle_lambda_weight()` is the same as running `ListMLELambdaWeight()`?

If yes, then we probably don't need to include this function. I recognize that it's inspired by the Tensorflow code, by the way.

Reply:
Your right. The lambdaweight can be initialized in two ways: Using `create_p_list_mle_lambda_weight()` and `ListMLELambdaWeight()`. It would be better to delete `create_p_list_mle_lambda_weight` function.

Reply:
https://github.com/tomaarsen/sentence-transformers/pull/6/commits/12f1c71dd1409d865a810c80f2b0b629b35c3ef6